text_extensions_for_pandas/array/arrow_conversion.py

```diff
@@ -133,10 +132,15 @@ def span_to_arrow(char_span: SpanArray) -> pa.ExtensionArray:
     begins_array = pa.array(char_span.begin)
     ends_array = pa.array(char_span.end)
 
-    typ = ArrowSpanType(begins_array.type, char_span.target_text)
+    # Create a dictionary array from StringTable used in this span
+    dictionary = pa.array(list(char_span._string_table.things))
+    target_text_dict_array = pa.DictionaryArray.from_arrays(char_span._text_ids, dictionary)
```

Comment:
This uses the private `char_span._string_table` and `._text_ids`. It didn't seem like we really needed accessors for these so I left it like this.

Reply:
Well the dictionary in pyarrow is just an ordinary array, we just need to make sure that the indices match those in the `_text_ids` array, which I think they do. I believe I should unbox the "thing" first though.